src/pages/ReimbursementAccount/CompanyStep.js

```diff
@@ -174,123 +221,94 @@ class CompanyStep extends React.Component {
                         <ExpensiTextInput
                             label={this.props.translate('companyStep.legalBusinessName')}
                             containerStyles={[styles.mt4]}
-                            onChangeText={companyName => this.setValue({companyName})}
+                            onChangeText={value => this.clearErrorAndSetValue('companyName', value)}
                             value={this.state.companyName}
                             disabled={shouldDisableCompanyName}
+                            errorText={this.getErrorText('companyName')}
                         />
                         <ExpensiTextInput
                             label={this.props.translate('common.companyAddress')}
                             containerStyles={[styles.mt4]}
-                            onChangeText={(addressStreet) => {
-                                if (error === this.props.translate('bankAccount.error.addressStreet')) {
-                                    hideBankAccountErrors();
-                                }
-                                this.setValue({addressStreet});
-                            }}
+                            onChangeText={value => this.clearErrorAndSetValue('addressStreet', value)}
                             value={this.state.addressStreet}
-                            errorText={error === this.props.translate('bankAccount.error.addressStreet')
-                                ? this.props.translate('bankAccount.error.addressStreet')
-                                : ''}
+                            errorText={this.getErrorText('addressStreet')}
                         />
                         <Text style={[styles.mutedTextLabel, styles.mt1]}>{this.props.translate('common.noPO')}</Text>
                         <View style={[styles.flexRow, styles.mt4]}>
                             <View style={[styles.flex2, styles.mr2]}>
                                 <ExpensiTextInput
                                     label={this.props.translate('common.city')}
-                                    onChangeText={addressCity => this.setValue({addressCity})}
+                                    onChangeText={value => this.clearErrorAndSetValue('addressCity', value)}
                                     value={this.state.addressCity}
+                                    errorText={this.getErrorText('addressCity')}
                                 />
                             </View>
                             <View style={[styles.flex1]}>
                                 <StatePicker
-                                    onChange={addressState => this.setValue({addressState})}
+                                    onChange={value => this.clearErrorAndSetValue('addressState', value)}
                                     value={this.state.addressState}
+                                    hasError={this.getErrors().addressState}
```

Comment:
Are we going to add a label for this one?

Reply:
I was just putting the red outline because there is little space under the `StatePicker`, do you want me to test how it would look with the error text there?

Reply:
Could we show the inline message in the space here?

![2021-09-13_11-19-48](https://user-images.githubusercontent.com/32969087/133157914-60f327bd-00ef-4ecf-904b-668c78ea83ff.png)

Any thoughts @shawnborton ? Or should we just let this one go?